library/oneview_server_profile.py

```diff
@@ -330,8 +330,10 @@ def __present(self):
                 if self.data.get('serverHardwareUri') is None and server_hardware_uri_exists:
                     self.data['serverHardwareUri'] = None
 
-            # Auto assigns a Server Hardware to Server Profile if auto_assign_server_hardware is True and no SH uris exist
-            if not self.current_resource.data.get('serverHardwareUri') and not self.data.get('serverHardwareUri') and self.auto_assign_server_hardware:
+            # Auto assigns a Server Hardware to Server Profile if auto_assign_server_hardware is True and no SH uris/enclosure uri and bay exist
+            if not self.current_resource.data.get('serverHardwareUri') and not self.data.get('serverHardwareUri') and self.auto_assign_server_hardware \
+                and not self.current_resource.data.get('enclosureUri') and not self.current_resource.data.get('enclosureBay') \
```

Comment:
Should we pass serverHardwareUri explicitly if enclosureBay and enclosureUri exist?

Reply:
not required.. This if condition works if serverHardwareUri and enclosureUri and bay are false and then goes inside. Did i miss anything..?

Reply:
No, I just wanted to confirm OneView is picking the server hardware if enclsureUri and Bay passed.

Reply:
Just to clarify the above comment, Server profile creation will be successful if we pass either serverHardwareUri (or) enclosureUri and bay number.